promql/parse.go

```diff
@@ -354,6 +368,9 @@ func (p *parser) expr() Expr {
 
 		// Parse the next operand.
 		rhs := p.unaryExpr()
+		if rhs == nil {
+			p.errorf("could not find right-hand side of binary expression")
```

Comment:
keep it simple: "could not find" -> "missing" (and then s/of/in/ sounds better)

"missing right-hand side in binary expression"